File: rcctr/code/ruby/groceries/public/javascripts/application.js
/***
 * Excerpted from "Continuous Testing",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material, 
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose. 
 * Visit http://www.pragmaticprogrammer.com/titles/rcctr for more book information.
***/
// Place your application-specific JavaScript functions and classes here
// This file is automatically included by javascript_include_tag :defaults

$.ajaxSetup({
  'beforeSend': function(xhr) { xhr.setRequestHeader('Accept', 'text/javascript'); }
});

modit('app', function() {
  function start() {}

  this.exports(start);
});

modit('app', function() {
  function start() {
    $('body').append("<div id='titlebar'/>");
  }

  this.exports(start);
});

modit('app', function() {
  function createToolbar() {
    $('body')
      .append($("<div id='titlebar'/>")
        .click(function() {
          $(this).toggle();
        })
     );
  }

  function start() {
    createToolbar();
    $('#shopping-list').each(function(i, element) {
      app.view.shoppingList(element);
    });
  }

  this.exports(start);
});

$(window.document).ready(app.start);

File: elixir16/code/tasks/my_app/lib/my_app/application.ex
#---
# Excerpted from "Programming Elixir ≥ 1.6",
# published by The Pragmatic Bookshelf.
# Copyrights apply to this code. It may not be used to create training material,
# courses, books, articles, and the like. Contact us if you are in doubt.
# We make no guarantees that this code is fit for any purpose.
# Visit http://www.pragmaticprogrammer.com/titles/elixir16 for more book information.
#---
defmodule MyApp.Application do
  @moduledoc false

  use Application

  def start(_type, _args) do
    children = [
      { MyApp.MyTask, 123 }
    ]

    opts = [strategy: :one_for_one, name: MyApp.Supervisor]
    Supervisor.start_link(children, opts)
  end
end
